docs/implement-check.md

```diff
@@ -0,0 +1,70 @@
+# Integrating Custom Check Into The Tool
+
+The Static Analysis Tool provides the necessary infrastructure for custom check implementations using the API of FindBugs, Checkstyle and PMD. 
+
+Helpful links when writing a custom check for the first time may be:
+
+- for [Checkstyle](http://checkstyle.sourceforge.net/writingchecks.html);
+- for [PMD](http://pmd.sourceforge.net/pmd-4.3.0/howtowritearule.html);
+- for [FindBugs](https://www.ibm.com/developerworks/library/j-findbug2/).
+
+In this guide we will use the Checkstyle API, because it is easy to use, supports different file extensions and languages. 
+
+We assume that you have already installed the Eclipse IDE and Maven.

```

Comment:
A good approach here will be to add links to a pages of Eclipse and Maven OR to provide links to ESH or openHAB tutorials that can help the user to install Eclipse and Maven.